### PR TITLE
Resolves LightBulbVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the LightBulbVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Light/Visualizers/LightBulbVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using Content.Shared.Light;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Light.Visualizers;
+
+[RegisterComponent]
+public sealed class LightBulbVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/LightBulbVisualizerComponent.cs:7]: merge into LightBulbComponent

File: Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+using Content.Shared.Light;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Light.Visualizers;
+
+public sealed class LightBulbVisualizerSystem : VisualizerSystem<LightBulbVisualizerComponent>
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs:6]: merge into LightBulbSystem

File: Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{
+    protected override void OnAppearanceChange(EntityUid uid, LightBulbVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+
+        // update sprite state
+        if (AppearanceSystem.TryGetData(uid, LightBulbVisuals.State, out LightBulbState state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs:14]: generic

File: Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {
+                case LightBulbState.Normal:
+                    args.Sprite.LayerSetState(0, "normal");
+                    break;
+                case LightBulbState.Broken:
+                    args.Sprite.LayerSetState(0, "broken");
+                    break;
+                case LightBulbState.Burned:
+                    args.Sprite.LayerSetState(0, "burned");
+                    break;
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs:26]: unhardcode

File: Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        }
+
+        // also update sprites color
+        if (AppearanceSystem.TryGetData(uid, LightBulbVisuals.Color, out Color color, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs:31]: generic

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Client/Light/Visualizers/LightBulbVisualizerSystem.cs]: No such system existed in shared or client. Had to move LightBulbComponent to shared.

File: Content.Shared/Light/Component/SharedLightBulbComponent.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+    #endregion Appearance
+}
+
+[Serializable, NetSerializable]
+public enum LightBulbState : byte
+{
+    Normal,
+    Broken,
+    Burned,
```
Comment (AJCM-git) [inline Content.Shared/Light/Component/SharedLightBulbComponent.cs:95]: i think the burned state is useless?

File: Content.Shared/Light/Component/SharedLightBulbComponent.cs [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Light.Component;
+
+/// <summary>
+/// Component that represents a light bulb. Can be broken, or burned, which turns them mostly useless.
+/// TODO: Breaking and burning should probably be moved to another component eventually.
+/// </summary>
+[RegisterComponent]
+public sealed class LightBulbComponent : Robust.Shared.GameObjects.Component
```
Comment (AJCM-git) [inline Content.Shared/Light/Component/SharedLightBulbComponent.cs:11]: rename the file accordingly (also VV fields)  (also doesnt this need NetworkedComponent?)

**Review by AJCM-git — CHANGES_REQUESTED** [review]
This introduces a bug where clicking a broken lightbulb inside a light doesnt return the item

File: Content.Shared/Light/Component/SharedLightBulbComponent.cs (hunk 6 above)
Comment (TemporalOroboros) [reply Content.Shared/Light/Component/SharedLightBulbComponent.cs:95]: Looks like it, yes.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**